app/src/main/java/com/github/se/studentconnect/ui/screen/profile/ProfileScreen.kt

```
@@ -0,0 +1,123 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.github.se.studentconnect.model.friends.FriendsRepositoryProvider
+import com.github.se.studentconnect.repository.UserRepository
+import com.github.se.studentconnect.repository.UserRepositoryFirestore
+import com.github.se.studentconnect.ui.profile.ProfileScreenViewModel
+import com.github.se.studentconnect.ui.profile.components.ProfileHeader
+import com.google.firebase.firestore.FirebaseFirestore
+
+/**
+ * Profile screen showing user information and stats. Main profile view for the StudentConnect app.
+ *
+ * @param currentUserId The ID of the current user (default for demo purposes)
+ * @param userRepository Repository for user data operations
+ * @param viewModel ViewModel for profile screen
+ * @param onNavigateToSettings Callback to navigate to settings/edit profile screen
+ * @param onNavigateToUserCard Callback to navigate to user card screen
+ * @param modifier Modifier for the composable
+ */
+@Composable
+fun ProfileScreen(
+    currentUserId: String = "mock_user_123",
```

Screens should not default to mock data. This can hide bugs during navigation. Use explicit parameters or preview-only overloads.